libpod/boltdb_state.go

```diff
@@ -115,6 +120,38 @@ func (s *BoltState) Refresh() error {
 		err = idBucket.ForEach(func(id, name []byte) error {
 			ctrBkt := ctrsBucket.Bucket(id)
 			if ctrBkt == nil {
+				// It's a pod
+				podBkt := podsBucket.Bucket(id)
+				if podBkt == nil {
+					// This is neither a pod nor a container
+					// Error out on the dangling ID
+					return errors.Wrapf(ErrInternal, "id %s is not a pod or a container", string(id))
+				}
+
+				// Get the state
+				stateBytes := podBkt.Get(stateKey)
+				if stateBytes == nil {
+					return errors.Wrapf(ErrInternal, "pod %s missing state key", string(id))
+				}
+
+				state := new(podState)
+
+				if err := json.Unmarshal(stateBytes, state); err != nil {
+					return errors.Wrapf(err, "error unmarshalling state for pod %s", string(id))
+				}
+
+				// Clear the CGroup path
+				state.CgroupPath = ""
```

Comment:
Reset runs after the system is restarted. CGroups won't survive system restart, so we clear the pod cgroup here, and then recreate it as the pod refreshes.

Reply:
With cgroupfs, this ends up being mostly a no-op, as we don't need to create the pod cgroup ourself, the first container to launch will make it for it. It will matter more with systemd cgroups, though.